### PR TITLE
java version checker targetCompatibility Java 9

### PR DESCRIPTION
**Pull request by ChrisHegarty**
I would like to bump the targetCompatibility of the java version
checker, from 7 to 9. I understand that we want this code targeting the
lowest possible byte code version, but 7 is very very old at this point.
I chose 9 as that is the minimum required to support Java Module
declarations, which we use over on the modules branch.

### CODE REVIEW

**Comment by ChrisHegarty** [comment]
TL;DR I propose to drop/close this PR (without merging)

> Will support for JPMS _matter_ here?

From the view point of the version checker alone, then java modules support does not really matter. But version checker is used from other modular code, namely the cli tools, therefore it must be modularized so that the cli tools module can `require` it.

> The other side of this is that Java 8 is still _very_ common and in those cases folks will get a very nasty error about unsupported class bytecode versions rather than our nicely formatted response. What do we _lose_ by keeping this at Java 8?

Yeah, this is a good point. The errors are less than idea! [1][2]. Notice that we cannot use newer command line options either - which is obvious now that we've thought about it.

Since the java version checker project has no dependencies beyond the JDK, then it's a relatively straightforward matter to compile the code with the existing target of 7, and the module-info with a target of 9. This will allow for the jar artifact to be run as 1) non-modular code on the class path (during startup), and 2) modular-code when being used as a dependency of a cli tool.  Note: this builds upon the fine design that [module artifacts][ma] can be put on the ordinary class path, in which case the module-info.class file is ignored. Additionally, the launchers should not use options that are not supported in ancient java versions (this is a specific note-to-self for the modules branch).

I implemented the above in the modules branch, see 412d2e9e, and it seems to work very nicely indeed. 

[ma]: https://openjdk.java.net/projects/jigsaw/spec/sotms/#module-artifacts
[com]: https://github.com/elastic/elasticsearch/pull/81066/commits/412d2e9eec8e724c845a90b32233de1628feef3d

[1]
```
$ ./distribution/archives/darwin-aarch64-tar/build/install/elasticsearch-8.2.0-SNAPSHOT/bin/elasticsearch
warning: ignoring JAVA_HOME=/Users/chegar/binaries/jdk-17.0.2.jdk/Contents/Home/; using ES_JAVA_HOME
Unrecognized option: --module-path
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

[2]
```
$ ./distribution/archives/darwin-aarch64-tar/build/install/elasticsearch-8.2.0-SNAPSHOT/bin/elasticsearch
warning: ignoring JAVA_HOME=/Users/chegar/binaries/jdk-17.0.2.jdk/Contents/Home/; using ES_JAVA_HOME
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.UnsupportedClassVersionError: org/elasticsearch/tools/java_version_checker/JavaVersionChecker has been compiled by a more recent version of the Java Runtime (class file version 53.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:495)
```
